### PR TITLE
dialect/entsql: add support for index-type annotation 

### PR DESCRIPTION
**Pull request by a8m**
In future PRs, we can add support for `Search` predicates to simplify constructing the WHERE expression manually. Also, we should consider add the common index types to entsql:

```go
entsql.FullText()
entgql.GIN()
...
```

### CODE REVIEW

File: dialect/sql/schema/mysql.go [hunk 1]
```diff
@@ -951,5 +951,22 @@ func (d *MySQL) atIndex(idx1 *Index, t2 *schema.Table, idx2 *schema.Index) error
...
+
+func indexType(idx *Index, d string) (string, bool) {
+	ant := idx.Annotation
+	if ant == nil {
+		return "", false
+	}
+	if ant.Type != "" {
+		return ant.Type, true
+	}
+	if ant.Types != nil && ant.Types[d] != "" {
```
Comment (masseelch) [inline dialect/sql/schema/mysql.go:968]: I'd expect `Types` to take precedence. Understood `Type` as default. And `Types` to override eg SQLite

**Review by yonidavidson — APPROVED** [review]

File: entc/integration/migrate/entv2/schema/user.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"entgo.io/ent"
+	"entgo.io/ent/dialect"
```
Comment (masseelch) [inline entc/integration/migrate/entv2/schema/user.go:11]: Indention seems wrong

File: dialect/sql/schema/mysql.go (hunk 1 above)
Comment (a8m) [reply dialect/sql/schema/mysql.go:965]: done

File: entc/integration/migrate/entv2/schema/user.go (hunk 2 above)
Comment (a8m) [reply entc/integration/migrate/entv2/schema/user.go:11]: why?

**Pull request merged**